mobly/suite_runner.py

```diff
@@ -70,7 +107,89 @@ def run_suite(test_classes, argv=None):
       help='A list of test classes and optional tests to execute.')
   if not argv:
     argv = sys.argv[1:]
-  args = parser.parse_args(argv)
+  return parser.parse_args(argv)
+
+
+def _find_suite_class():
+  """Finds the test suite class in the current module.
+
+  Walk through module members and find the subclass of BaseSuite. Only
+  one subclass is allowed in a module.
+
+  Returns:
+      The test suite class in the test module.
+  """
+  test_suites = []
+  main_module_members = sys.modules['__main__']
+  for _, module_member in main_module_members.__dict__.items():
+    if inspect.isclass(module_member):
+      if issubclass(module_member, base_suite.BaseSuite):
+        test_suites.append(module_member)
+  if len(test_suites) != 1:
+    logging.error('Expected 1 test class per file, found %s.',
+                  [t.__name__ for t in test_suites])
+    sys.exit(1)
+  return test_suites[0]
+
+
+class _SuiteRunner(object):
+  """Class used to execute a Mobly suite in google3."""
```

Comment:
I guess this is more like a reference implementation of what a suite runner could be like?